community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -786,9 +786,14 @@ public SchemaWrite schemaWrite()
     }
 
     @Override
-    public StatementLocks locks()
+    public org.neo4j.internal.kernel.api.Locks locks()
     {
-       return statementLocks;
+       return operations.locks();
```

Comment:
that's a strategic move to mention kernel team to take a look at this one;
just mainly fyi this change will affect some of our customers with their custom code around locks so i would ask to provide a good description of what moves where compare to previous code version

Reply:
I don't understand the first sentence, if you're meaning that I intentionally didn't add kernel as a labels - I did not. I merely forgot.

This PR only moves code around and should not affect customers code around locks. That method your are commenting on was added here https://github.com/neo4j/neo4j/pull/10418/commits/5be9251a89f670fe343da5cc2cfafb264bb1ba74#diff-e846f0ece315673d2137c213a8b1c823R715 and was not there in earlier versions

Reply:
do know the intentions, trying to stop reading people's mind, at least at work;

we know about customers that get underlying transaction implementation in this case KernelTransactionImplementaion and use locks from it (previously known as StatementLocks interface). 
Current PR changes that internal contract and those custom extensions will be affected by this change.
Its not like this should stop your redesign or way of your thoughts. I'm just asking for a more clear description of this change so they/we later one will be able to adapt that code, bu just reading corresponding PR desription

Reply:
But it is not changing it, transactions still have an internal statementLocks? The `StatementLock` interface is unchanged, this just removed a temporary hack where `StatementLocks` extended the empty interface `Locks`.

Reply:
no, but you haven't been able to do that of any released version of neo4j that I know of. You could do it in some of the 3.4-alphas though.  
The `locks` method is part of the new `Transaction` API that was added in 3.4, it returns a `Locks` which up to this PR was a completely empty interface. For some reason in some intermediate step in 3.4 we made `StatementLocks` extend `Locks` and return the statementLocks from this method instead. 

Main point: `((KernelTransactionImplementation)tx).locks().pessimistic()` has - excluding one or maybe two alphas - never been possible in any released version of neo4j.

Reply:
ah ok now I see, sorry for confusion

Reply:
np :)